### PR TITLE
Fix panic

### LINKED ISSUE #16: panic: interface conversion: ast.Expr is *ast.SelectorExpr, not *ast.Ident
The linter panics on the following code
```go
func Example(service protoreflect.ServiceDescriptor) {
	for i := 0; i < service.Methods().Len(); i++ {
		print(i)
	}
}
```

```
panic: interface conversion: ast.Expr is *ast.SelectorExpr, not *ast.Ident

goroutine 70 [running]:
github.com/ckaznocha/intrange.findNExpr({0x138cfc8?, 0xc000216340?})
        external/gazelle~~go_deps~com_github_ckaznocha_intrange/intrange.go:237 +0x13a
github.com/ckaznocha/intrange.run.check.func1({0x138c4e0?, 0xc0002163c0})
        external/gazelle~~go_deps~com_github_ckaznocha_intrange/intrange.go:95 +0x1f4
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0xc000010228, {0xc00022ed18?, 0x1538860?, 0xc000052908?}, 0xc00022ed08)
        external/gazelle~~go_deps~org_golang_x_tools/go/ast/inspector/inspector.go:82 +0x8f
github.com/ckaznocha/intrange.run(0xc0001a8270)
        external/gazelle~~go_deps~com_github_ckaznocha_intrange/intrange.go:47 +0x129
main.(*action).execOnce(0xc0001ade00)
        external/rules_go~/go/tools/builders/nogo_main.go:354 +0x8b7
sync.(*Once).doSlow(0x0?, 0xc000123200?)
        GOROOT/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        GOROOT/src/sync/once.go:65
main.(*action).exec(...)
        external/rules_go~/go/tools/builders/nogo_main.go:281
main.execAll.func1(0x0?)
        external/rules_go~/go/tools/builders/nogo_main.go:275 +0x72
created by main.execAll in goroutine 1
        external/rules_go~/go/tools/builders/nogo_main.go:273 +0x48
```

### PR DESCRIPTION
**Pull request by maratori**
Closes #16

### CODE REVIEW

File: testdata/go.mod [hunk 1]
```diff
@@ -1,3 +1,5 @@
 module github.com/ckaznocha/intrange/testdata
 
-go 1.21
+go 1.22
```
Comment (ckaznocha) [inline testdata/go.mod:3]: Please revert this line. To be compatible with golangci-lint it needs to support go1.21.

Comment (maratori) [reply testdata/go.mod]: Didn't know that. Done.

File: testdata/main.go [hunk 2]
```diff
@@ -187,3 +191,10 @@ func main() {
 		t2.m[4] = 4
 	}
 }
+
+// https://github.com/ckaznocha/intrange/issues/16
+func issue16(service protoreflect.ServiceDescriptor) {
```
Comment (ckaznocha) [inline testdata/main.go:196]: I like the way you set up this test, I may update the others to be similar!

File: testdata/go.mod (hunk 1 above)
Comment (ckaznocha) [reply testdata/go.mod]: Oh, this was the test go.mod, that probably would have been fine. Sorry, that's what I get for doing pr reviews on my phone.

**Review by ckaznocha — APPROVED** [review]
Thanks!

**Pull request merged**